ihp-ide/IHP/SchemaCompiler.hs

```diff
@@ -857,7 +906,30 @@ compileDataTypePattern :: (?schema :: Schema) => CreateTable -> Text
 compileDataTypePattern table@(CreateTable { name }) = tableNameToModelName name <> " " <> unwords (table |> dataFields |> map fst)
 
 compileTypePattern :: (?schema :: Schema) => CreateTable -> Text
-compileTypePattern table@(CreateTable { name }) = tableNameToModelName name <> "' " <> unwords (dataTypeArguments table)
+compileTypePattern table@(CreateTable { name, inherits }) = tableNameToModelName name <> "' " <> dataTypeCompiled
+    where
+        dataTypeCompiled = if null parentTypeArguments
+            then currentTypeArguments
+            else currentTypeArguments <> " " <> parentTypeArguments
+
+        currentTypeArguments = dataTypeArguments table |> unwords
+
+        modelName = tableNameToModelName name
+
+        -- @todo: Find a better way.
+        colName = modelName |> pluralize |> Text.toLower
+
+        parentTypeArguments :: Text
```

Comment:
@mpscholten I continue to hack through this. The latest error is this

![image](https://github.com/user-attachments/assets/1dd387fa-9a0d-42e6-8618-fb6d69c039ce)

Here's the generated code. Are you able to spot what I'm doing wrong in terms of types?

```haskell
data PostRevision' postId userId = PostRevision {title :: Text, body :: Text, userId :: userId, id :: (Id' "post_revisions"), postId :: postId, meta :: MetaBag} deriving (Eq, Show)

type instance PrimaryKey "post_revisions" = UUID

type PostRevision = PostRevision' (Id' "posts")

type instance GetTableName (PostRevision' _) = "post_revisions"
type instance GetModelByTableName "post_revisions" = PostRevision

instance Default (Id' "post_revisions") where def = Id def

instance () => Table (PostRevision' postId userId) where
    tableName = "post_revisions"
    tableNameByteString = Data.Text.Encoding.encodeUtf8 "post_revisions"
    columnNames = ["id","post_id","id","title","body","userId","postRevisions"]
    primaryKeyColumnNames = ["id"]
    primaryKeyConditionForId (Id (id)) = toField id
    {-# INLINABLE primaryKeyConditionForId #-}
```

Reply:
It looks like the `type PostRevision` is wrong

```haskell
-- Current
type PostRevision = PostRevision' (Id' "posts")

-- Should be:
type PostRevision = PostRevision' (Id' "posts") (Id' "users")
```